src/colorizer/ColorRamp.ts

```diff
@@ -61,20 +62,21 @@ export default class ColorRamp {
     if (this.colorStops.length < 2) {
       ctx.fillStyle = `#${this.colorStops[0].getHexString()}`;
       ctx.fillRect(0, 0, width, height);
-    } else if (this.type === ColorRampType.LINEAR) {
-      const gradientWidth = vertical ? 0 : width;
-      const gradientHeight = vertical ? height : 0;
-      const gradient = ColorRamp.linearGradientFromColors(ctx, this.colorStops, gradientWidth, gradientHeight);
-      ctx.fillStyle = gradient;
-      ctx.fillRect(0, 0, width, height);
-    } else {
+    } else if (this.type === ColorRampType.CATEGORICAL) {
       // Draw as hard stop gradients
       const stops = this.colorStops.slice(0, DISPLAY_GRADIENT_MAX_STOPS);
       const step = width / stops.length;
       stops.forEach((color, idx) => {
         ctx.fillStyle = `#${color.getHexString()}`;
         ctx.fillRect(Math.floor(step * idx), 0, Math.ceil(step), height);
       });
+    } else {
+      // All other ramp types are linear gradients
+      const gradientWidth = vertical ? 0 : width;
+      const gradientHeight = vertical ? height : 0;
+      const gradient = ColorRamp.linearGradientFromColors(ctx, this.colorStops, gradientWidth, gradientHeight);
+      ctx.fillStyle = gradient;
+      ctx.fillRect(0, 0, width, height);
```

Comment:
Flipped around this conditional so that the Linear case is in the else branch, but all code is unchanged.